### PR TITLE
Add support for Casbin AddPolicies and RemovePolicies

### PR DESCRIPTION
**Pull request by dallen42**

### CODE REVIEW

**Comment by nodece** [comment]
@dallen42 Can we support transactions? An error occurred When deleting or adding policy , we need to rollback.

**Comment by dallen42** [comment]
@nodece agreed, was not aware of that Gorm functionality. Updated to use transactions w rollback on errors

File: adapter.go [hunk 1]
```diff
@@ -385,6 +385,34 @@ func (a *Adapter) RemovePolicy(sec string, ptype string, rule []string) error {
 	return err
 }
 
+// AddPolicies adds multiple policy rules to the storage.
+func (a *Adapter) AddPolicies(sec string, ptype string, rules [][]string) error {
+	return a.db.Transaction(func(tx *gorm.DB) error {
+		for _, rule := range rules {
+			line := a.savePolicyLine(ptype, rule)
+			if err := tx.Create(&line).Error; err != nil {
+				tx.Rollback()
```
Comment (nodece) [inline adapter.go:394]: returns `err` will auto rollback.

File: adapter.go [hunk 2]
```diff
@@ -385,6 +385,34 @@ func (a *Adapter) RemovePolicy(sec string, ptype string, rule []string) error {
...
+	})
+}
+
+// RemovePolicy removes multiple policy rules from the storage.
+func (a *Adapter) RemovePolicies(sec string, ptype string, rules [][]string) error {
+	return a.db.Transaction(func(tx *gorm.DB) error {
+		for _, rule := range rules {
+			line := a.savePolicyLine(ptype, rule)
+			if err := a.rawDelete(a.db, line); err != nil { //can't use db.Delete as we're not using primary key http://jinzhu.me/gorm/crud.html#delete
+				tx.Rollback()
```
Comment (nodece) [inline adapter.go:408]: the same as above.

File: adapter.go (hunk 1 above)
Comment (dallen42) [reply adapter.go]: @nodece fixed

File: adapter.go (hunk 2 above)
Comment (dallen42) [reply adapter.go]: @nodece fixed

**Comment by nodece** [comment]
Hi @hsluoyz , could you review the PR?

**Comment by hsluoyz** [comment]
lgtm

**Pull request merged**